### ISSUE TITLE
StatefulSet: update forbidden because of PVC storage units

### ISSUE THREAD

**Issue by derekperkins** [kubernetes/kubernetes#66137]
/kind bug

When deploying a StatefulSet, I included a PVC that specified `2048Gi` for the storage capacity. `kubectl apply` worked perfectly. Running `kubectl apply` again without making any changes failed with the error:

```
The StatefulSet "ss" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden.
```

After comparing the server-side yaml to my original, I noticed that it had changed the storage capacity of my PVC to `2Ti`. Changing my original from `2048Gi` to `2Ti` allowed me to then successfully run `kubectl apply`.

There is clearly a string representation being used to compare those changes rather than converting to bytes and comparing the numeric representation.

**How to reproduce it (as minimally and precisely as possible)**:
```
  volumeClaimTemplates:
  - metadata:
      name: data
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 2048Gi
```

**Environment**:
- Kubernetes version (use `kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T22:29:25Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: Azure AKS

**Comment by derekperkins** [comment]
@kubernetes/sig-storage-bugs

**Comment by msau42** [comment]
cc @k8s-mirror-api-machinery-bugs 

This is probably an issue for any resource.Quantity type.

**Comment by liggitt** [comment]
fix is in #66165 with picks open to 1.11, 1.10, 1.9
